client/java/src/main/java/io/openlineage/client/utils/OpenLineageEnvParser.java

```diff
@@ -0,0 +1,154 @@
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import io.openlineage.client.Environment;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class OpenLineageEnvParser {
+
+  private static final ObjectMapper objectMapper = new ObjectMapper();
+  private static final String PREFIX = "OPENLINEAGE__";
+
+  /**
+   * Parses all environment variables starting with OPENLINEAGE__ into a single nested JSON
+   * structure.
+   *
+   * @return A single nested JSON structure with all matching environment variables.
+   * @throws JsonProcessingException
+   */
+  public static String parseAllOpenLineageEnvVars() throws JsonProcessingException {
+    ObjectNode root = objectMapper.createObjectNode();
+
+    // Filter all environment variables starting with 'OPENLINEAGE__'
+    Map<String, String> env = Environment.getAllEnvironmentVariables();
+    env.entrySet().stream()
+        .filter(entry -> entry.getKey().startsWith(PREFIX))
+        .sorted(Map.Entry.comparingByKey(Comparator.reverseOrder()))
+        .map(
+            entry -> {
+              String envKey = entry.getKey();
+              String envValue = entry.getValue();
+              try {
+                // Parse the individual environment variable into a nested structure
+                return parseEnvToNestedJson(envKey, envValue);
+              } catch (Exception e) {
+                log.warn("Failed to parse environment variable: {}", envKey);
+                return null;
+              }
+            })
+        .filter(parsedJson -> parsedJson != null) // Filter out any failed parsing attempts
+        .forEach(parsedJson -> merge(root, parsedJson));
+    String jsonString = objectMapper.writerWithDefaultPrettyPrinter().writeValueAsString(root);
+    return jsonString;
+  }
+
+  /**
+   * Parses a single environment variable key and value into a nested JSON structure.
+   *
+   * @param envKey The environment variable key.
+   * @param envValue The environment variable value.
+   * @return A nested JSON structure based on the key.
+   */
+  private static ObjectNode parseEnvToNestedJson(String envKey, String envValue) throws Exception {
+    // Remove the 'OPENLINEAGE__' prefix
+    String strippedKey = envKey.substring(PREFIX.length());
+
+    // Split the remaining key on '__' into a List<String>
+    List<String> keyParts =
+        Stream.of(strippedKey.split("__"))
+            .map(String::toLowerCase) // Lowercase the keys if needed
+            .map(OpenLineageEnvParser::convertSnakeCaseToCamelCase)
+            .collect(Collectors.toList());
+
+    // Create the final value node. If it's valid JSON, parse it; otherwise, treat
+    // it as a string.
+    JsonNode valueNode;
+    try {
+      valueNode = objectMapper.readTree(envValue);
+    } catch (Exception e) {
```

Comment:
I am not sure if I understand that fully.
I think readTree can throw
```java
JsonProcessingException, JsonMappingException
```

Can we lower the error scope ?

Reply:
Lowered to `JsonProcessingException`.